### DISCUSSION TITLE
Is it okay to use function as key?

### DISCUSSION THREAD

**Discussion by chiawendt** [vercel/swr#1948] [Q&A] [answered]
In my project our API endpoints are accessed as (constant) functions likes `listBooks`, `getBook`, and these functions are used like

```js
import {getBook} from '../api/book'

function Foo() {
  const id = ...;
  const {data} = useSWR([getBook, id], () => {
    return getBook(id)
  })
  ...
}
```

Is it okay to use function as key like this, as long as it is inside an array?

**Comment by koba04** [comment]
I'm not sure whether SWR officially supports this, but your example works fine if `getBook` always returns the same reference.

**Comment by shuding** [comment, marked as answer]
Yep just like @koba04 said, it's a valid use case as long as you don't directly pass a function.
